Core/src/Material/IntersectionMaterialAssigner.cpp

```diff
@@ -0,0 +1,116 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Material/IntersectionMaterialAssigner.hpp"
+
+#include "Acts/Surfaces/BoundaryCheck.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+
+namespace {
+
+std::vector<Acts::SurfaceIntersection> forwardOrderedIntersections(
+    const Acts::GeometryContext& gctx, const Acts::Vector3& position,
+    const Acts::Vector3& direction,
+    const std::vector<const Acts::Surface*>& surfaces) {
+  // First deal with the surface intersections
+  std::vector<Acts::SurfaceIntersection> sIntersections;
+  // Intersect the surfaces
+  for (auto& surface : surfaces) {
+    // Get the intersection
+    auto sMultiIntersection = surface->intersect(gctx, position, direction,
+                                                 Acts::BoundaryCheck(true));
+    // One solution, take it
+    if (sMultiIntersection.size() == 1u &&
+        sMultiIntersection[0u].status() >=
+            Acts::IntersectionStatus::reachable &&
+        sMultiIntersection[0u].pathLength() >= 0.0) {
+      sIntersections.push_back(sMultiIntersection[0u]);
+      continue;
+    }
+    if (sMultiIntersection.size() > 1u) {
+      // Multiple intersections, take the closest
+      auto closestForward = sMultiIntersection.closestForward();
+      if (closestForward.status() >= Acts::IntersectionStatus::reachable &&
+          closestForward.pathLength() > 0.0) {
+        sIntersections.push_back(closestForward);
+        continue;
+      }
+    }
+  }
+  // Sort the intersection along the pathlength
+  std::sort(sIntersections.begin(), sIntersections.end(),
+            &Acts::SurfaceIntersection::pathLengthOrder);
+  return sIntersections;
+}
+
+}  // namespace
+
+std::pair<std::vector<Acts::IAssignmentFinder::SurfaceAssignment>,
+          std::vector<Acts::IAssignmentFinder::VolumeAssignment>>
+Acts::IntersectionMaterialAssigner::assignmentCandidates(
+    const GeometryContext& gctx, const MagneticFieldContext& /*unused*/,
```

Comment:
```suggestion
    const GeometryContext& gctx, const MagneticFieldContext& /*mctx*/,
```

should the magnetic field context be dropped if it is not used here?

Reply:
It's needed for the propagator implementation of the Intersection assigner, so I guess we can leave it here.